src/worker.cc

```diff
@@ -802,6 +805,36 @@ void from_json(rapidjson::Document& doc, Options& options) {
     options.set_resample_distance(*resample_distance);
   }
 
+  // expansion action, default to isochrone
+  auto exp_action_str = rapidjson::get_optional<std::string>(doc, "/action");
+  Options::Action exp_action = Options::isochrone;
+  if (exp_action_str) {
+    if (!Options_ExpansionAction_Enum_Parse(*exp_action_str, &exp_action)) {
+      throw valhalla_exception_t(144, *exp_action_str);
+    }
+    options.set_expansion_action(exp_action);
+  } else if (options.action() == Options_Action_expansion) {
+    throw valhalla_exception_t(115, std::string("action"));
+  }
+
+  // expansion response properties
+  auto exp_props_req = rapidjson::get_child_optional(doc, "/expansion_props");
+  auto* exp_props_pbf = options.mutable_expansion_props();
+  Options::ExpansionProps exp_prop;
+  if (exp_props_req && exp_props_req->IsArray()) {
+    for (const auto& prop : exp_props_req->GetArray()) {
+      if (!valhalla::Options_ExpansionProps_Enum_Parse(std::string(prop.GetString()), &exp_prop)) {
+        throw valhalla_exception_t(168, std::string(prop.GetString()));
+      }
+      exp_props_pbf->Add(exp_prop);
+    }
+  }
```

Comment:
only throw on unsupported property keys; allow for no keys at all (ie empty `properties` array in the response, which doesn't seem to be explicitly forbidden according to spec, geojson.io is happy with `properties` array)